### PR TITLE
Update nearly all nuget packages and .Net to 8.0

### PR DESCRIPTION
**Pull request by trishorts**
.net 6 and 7 no longer have support, which prompted this move to .net 8.  To keep up w/ project dependencies and compatibilities, all nuget packages were attempted to be updated. 

I generated a local nuget package with this PR and put it into MetaMorpheus. I analyzed 6 Mann files with calibration, averaging, gptmd and search. Results loaded into MetaDraw successfully. I also performed a successful glyco search with test data. I don't see any difficulties integrating this into MetaMorpheus.

OxyPlot.Wpf COULD NOT BE UPDATED
SharpLearning.Optimization COULD NOT BE UPDATED

A close look at the impact of updating these nuget packages on MetaMorpheus and other downstream apps should be performed.

Updating NuUnit was imperfect. The Assert function usage has changed substantially. To avoid massive refactoring, Assert was recast as ClassicAssert. Going forward, we should attempt to use the new Nunit usage.

TargetFramework was updated to 8.0 for all projects in the properties menu of each project in the solution explorer.
The .nuspec was updated to .net 8.0

The github/workflows/dotnet.yml and release.yml were updated to .net 8

coverlet.collector in dotnet.yml updated to -v 6.0.2

### CODE REVIEW

**Comment by trishorts** [comment]
please see issue: https://github.com/smith-chem-wisc/mzLib/issues/781

File: mzLib/mzLib.nuspec [hunk 1]
```diff
@@ -22,8 +22,8 @@
         <dependency id="OpenMcdf.Extensions" version="2.2.1.3" />
         <dependency id="SharpLearning.Optimization" version="0.28.0" />
       </group>
-      <group targetFramework="net6.0-windows7.0">
-        <!--net6.0-windows7.0 target with OxyPlot-->
+      <group targetFramework="net8.0-windows8.0">
```
Comment (acesnik) [inline mzLib/mzLib.nuspec:25]: I think this can be `net8.0-windows`

File: mzLib/mzLib.nuspec [hunk 2]
```diff
@@ -37,48 +37,48 @@
...
+    <file src="Readers\bin\x64\Release\net8.0\Readers.dll" target="lib\net8.0" />
+    <file src="Readers\bin\x64\Release\net8.0\Thermo\ThermoFisher.CommonCore.BackgroundSubtraction.dll" target="lib\net8.0" />
+    <file src="Readers\bin\x64\Release\net8.0\Thermo\ThermoFisher.CommonCore.Data.dll" target="lib\net8.0" />
+    <file src="Readers\bin\x64\Release\net8.0\Thermo\ThermoFisher.CommonCore.MassPrecisionEstimator.dll" target="lib\net8.0" />
+    <file src="Readers\bin\x64\Release\net8.0\Thermo\ThermoFisher.CommonCore.RawFileReader.dll" target="lib\net8.0" />
+    <file src="UsefulProteomicsDatabases\bin\x64\Release\net8.0\UsefulProteomicsDatabases.dll" target="lib\net8.0" />
+    <file src="Transcriptomics\bin\x64\Release\net8.0\Transcriptomics.dll" target="lib\net8.0" />
+    <file src="Transcriptomics\bin\x64\Release\net8.0\Digestion\rnases.tsv" target="lib\net8.0" />
+    <!--net8.0-windows target with mzPlot-->
+    <file src="mzPlot\bin\x64\Release\net8.0-windows\mzPlot.dll" target="lib\net8.0-windows8.0" />
```
Comment (acesnik) [inline mzLib/mzLib.nuspec:62]: `net8.0-windows`

File: mzLib/mzPlot/mzPlot.csproj [hunk 3]
```diff
@@ -1,11 +1,12 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0-windows</TargetFramework>
+    <TargetFramework>net8.0-windows7.0</TargetFramework>
```
Comment (acesnik) [inline mzLib/mzPlot/mzPlot.csproj:4]: Does it work to delete the `7.0` here?

File: mzLib/TestFlashLFQ/TestFlashLFQ.csproj [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0-windows</TargetFramework>
+    <TargetFramework>net8.0-windows7.0</TargetFramework>
```
Comment (acesnik) [inline mzLib/TestFlashLFQ/TestFlashLFQ.csproj:4]: `net8.0-windows`?

**Review by acesnik — CHANGES_REQUESTED** [review]
For the packages that can't be updated, are they just not released in net8.0, yet? I wonder if you could post an issue on their repositories to queue that, so they can be updated without needing to investigate the downstream issues.

**Comment by trishorts** [comment]
> For the packages that can't be updated, are they just not released in net8.0, yet? I wonder if you could post an issue on their repositories to queue that, so they can be updated without needing to investigate the downstream issues.

It is not a problem of net8 compatibility. rather its a problem of older features in these nuget packages that are not supported in the newer release, which would necessitate changes to programs such as MM that use these old features. a focused PR to update each of these two remaining packages should be put on the todo list.

File: mzLib/TestFlashLFQ/TestFlashLFQ.csproj (hunk 4 above)
Comment (trishorts) [reply mzLib/TestFlashLFQ/TestFlashLFQ.csproj]: these were "replace all" errors. I think they are fixed now.

**Comment by trishorts** [comment]
may need to update the nuget website for the target. 
image

**Review by nbollis — APPROVED** [review]
lgtm

File: mzLib/mzLib.nuspec (hunk 1 above)
Comment (nbollis) [reply mzLib/mzLib.nuspec]: this change should be made to net8.0-windows7.0

File: mzLib/mzLib.nuspec (hunk 2 above)
Comment (nbollis) [reply mzLib/mzLib.nuspec]: end of these lines should be changed to windows7.0

File: mzLib/mzPlot/mzPlot.csproj (hunk 3 above)
Comment (nbollis) [reply mzLib/mzPlot/mzPlot.csproj]: I believe we need the windows7.0 here to display the plots in WPF through MetaDraw

**Review by nbollis — CHANGES_REQUESTED** [review]

**Review by nbollis — APPROVED** [review]

**Review by RayMSMS — APPROVED** [review]

**Review by acesnik — APPROVED** [review]

**Pull request merged**